actions/nix-build/action.yml

```
@@ -0,0 +1,142 @@
+name: nix-build
+description: "Builds all manifest models and uploads them to a specified cache"
+
+inputs:
+  # general inputs --------------------------------
+  add-packages:
+    description: "Additional packages to build on top of the declared check packages (space separated) - not used if setup-only=true"
+    required: false
+    default: ""
+
+  # cache inputs ----------------------------------
+  cache-url:
+    description: "Nix cache URL"
+    required: false
+    default: ""
+  cache-privkey:
+    description: "Nix cache binary private signing key - not used if setup-only=true"
+    required: false
+    default: ""
+
+  # aws inputs --------------------------
+  # see https://github.com/aws-actions/configure-aws-credentials/blob/main/action.yml
+  aws-region:
+    description: "AWS region"
+    required: false
+    default: us-west-2
+  role-to-assume:
+    description: The Amazon Resource Name (ARN) of the role to assume. Use the provided credentials to assume an IAM role and configure the Actions environment with the assumed role credentials rather than with the provided credentials.
+    required: false
+  role-duration-seconds:
+    description: Role duration in seconds. Default is one hour.
+    required: false
+  aws-access-key-id:
+    description: AWS Access Key ID. Provide this key if you want to assume a role using access keys rather than a web identity token.
+    required: false
+  aws-secret-access-key:
+    description: AWS Secret Access Key. Required if aws-access-key-id is provided.
+    required: false
+  aws-session-token:
+    description: AWS Session Token.
+    required: false
+
+  # grafana inputs --------------------------
+  metrics-enabled:
+    description: "enables grafana metrics"
+    required: false
+    default: true
+  metrics-job-name:
+    description: "grafana metrics job name"
+    required: false
+    default: nix-build
+  gc-host:
+    description: "grafana hostname"
+    required: false
+  gc-basic-auth:
+    description: "grafana basic auth"
+    required: false
+  gc-org-id:
+    description: "grafana org/tenant id"
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - name: AWS credentials
+      uses: aws-actions/configure-aws-credentials@v4
+      with:
+        aws-region: ${{ inputs.aws-region }}
+        role-to-assume: ${{ inputs.role-to-assume }}
+        role-duration-seconds: ${{ inputs.role-duration-seconds }}
+        aws-access-key-id: ${{ inputs.aws-access-key-id }}
+        aws-secret-access-key: ${{ inputs.aws-secret-access-key }}
```

```suggestion
        aws-access-key-id: ${{ inputs.aws-access-key-id }}
        aws-secret-access-key: ${{ inputs.aws-secret-access-key }}
        mask-aws-account-id: true
```

Don't think we need access keys (see earlier comment) and we should mask the AWS account ID so it doesn't appear in logs.

---

GitHub already does that for us, I believe - anything that comes from the secrets appears as `***` in the output of the action.